docs/source/conceptual/visual_language_models.md

```diff
@@ -0,0 +1,224 @@
+# Vision Language Models (VLM)
+
+## What is VLM?
+
+Visual Language Model (VLM) are models that consume both visual and textual inputs to generate text.
+
+These models are trained on multimodal data, which includes both images and text.
+
+VLMs can be used for a variety of tasks, such as image captioning, visual question answering, and more.
+
+<div class="flex justify-center">
+    <pre>placeholder for architecture diagram</pre>
+</div>
+
+With VLM, you can generate text from an image. For example, you can generate a caption for an image, answer questions about an image, or generate a description of an image.
+
+- **Image Captioning**: Given an image, generate a caption that describes the image.
+- **Visual Question Answering (VQA)**: Given an image and a question about the image, generate an answer to the question.
+- **Visual Dialog**: Given an image and a dialog history, generate a response to the dialog.
+- **Visual Data Extraction**: Given an image, extract information from the image.
+
+For example, given the image of a cat, a VLM can generate the caption "A cat sitting on a couch" or answer the question "What is the cat doing?" with "The cat is sitting on a couch."
+
+## How to use VLM?
+
+### VLM with Python
+
+To use VLM with Python, you can use the `huggingface_hub` library. The `InferenceClient` class provides a simple way to interact with the Inference API.
+
+This is the following image:
+
+<div class="flex justify-center">
+    <img
+        src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/transformers/rabbit.png"
+        width="400"
+    />
+</div>
+
+```python
+from huggingface_hub import InferenceClient
+
+client = InferenceClient("http://127.0.0.1:3000")
+image = "https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/transformers/rabbit.png"
+prompt = f"![]({image})What is this a picture of?\n\n"
+for token in client.text_generation(prompt, max_new_tokens=16, stream=True):
+    print(token)
+
+# This
+#  is
+#  a
+#  picture
+#  of
+#  an
+#  anth
+# rop
+# omorphic
+#  rab
+# bit
+#  in
+#  a
+#  space
+#  suit
+# .
+```
+
+Images can be passed as URLs or base64-encoded strings. The `InferenceClient` will automatically detect the image format.
+
+This is the following image:
+
+<div class="flex justify-center">
+    <img
+        src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/transformers/beaver.png"
+        width="400"
+    />
+</div>
+
+```python
+from huggingface_hub import InferenceClient
+import base64
+import requests
+import io
+
+client = InferenceClient("http://127.0.0.1:3000")
+
+url = "https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/transformers/beaver.png"
```

Comment:
maybe you could read it from directory this time which is the point of using base64 here

Reply:
updated to read image from file